src/mdl/holder.rs

```diff
@@ -0,0 +1,533 @@
+        None,
+    )
+    .map_err(|e| SessionError::Generic {
+        value: format!("Could not initialize session: {e:?}"),
+    })?;
+    let ble_ident = session
+        .ble_ident()
+        .map_err(|e| SessionError::Generic {
+            value: format!("Couldn't get BLE identification: {e:?}").to_string(),
+        })?
+        .to_vec();
+    let (engaged_state, qr_code_uri) =
+        session.qr_engagement().map_err(|e| SessionError::Generic {
+            value: format!("Could not generate qr engagement: {e:?}"),
+        })?;
+    Ok(MdlPresentationSession {
+        engaged: Mutex::new(Some(engaged_state)),
+        in_process: Mutex::new(None),
+        qr_code_uri,
+        ble_ident,
+    })
+}
+
+/// Begin the mDL presentation process for the holder by passing in the raw
+/// bytes of an Mdoc as a CBOR encoded Vec<u8>.
+///
+/// Initializes the presentation session for an ISO 18013-5 mDL and stores
+/// the session state object in the device storage_manager.
+///
+/// Arguments:
+/// mdoc_bytes: bytes of the Mdoc in CBOR encoded Vec<u8>
+/// uuid:       the Bluetooth Low Energy Client Central Mode UUID to be used
+///
+/// Returns:
+/// A Result, with the `Ok` containing a tuple consisting of an enum representing
+/// the state of the presentation, a String containing the QR code URI, and a
+/// String containing the BLE ident.
+///
+#[uniffi::export]
+pub async fn initialize_mdl_presentation_from_bytes(
+    mdoc_bytes: Vec<u8>,
+    key_alias: KeyAlias,
+    uuid: Uuid,
+) -> Result<MdlPresentationSession, SessionError> {
+    let mdoc = crate::credential::mdoc::Mdoc::from_cbor_encoded_document(mdoc_bytes, key_alias)
+        .map_err(|e| SessionError::Generic {
+            value: format!("Error retrieving MDoc from storage: {e:}"),
+        })?;
+    let drms = DeviceRetrievalMethods::new(DeviceRetrievalMethod::BLE(BleOptions {
+        peripheral_server_mode: None,
+        central_client_mode: Some(CentralClientMode { uuid }),
+    }));
+    let session = SessionManagerInit::initialise(
+        NonEmptyMap::new("org.iso.18013.5.1.mDL".into(), mdoc.document().clone()),
+        Some(drms),
+        None,
+    )
+    .map_err(|e| SessionError::Generic {
+        value: format!("Could not initialize session: {e:?}"),
+    })?;
+    let ble_ident = session
+        .ble_ident()
+        .map_err(|e| SessionError::Generic {
+            value: format!("Couldn't get BLE identification: {e:?}").to_string(),
+        })?
+        .to_vec();
+    let (engaged_state, qr_code_uri) =
+        session.qr_engagement().map_err(|e| SessionError::Generic {
+            value: format!("Could not generate qr engagement: {e:?}"),
+        })?;
+    Ok(MdlPresentationSession {
+        engaged: Mutex::new(Some(engaged_state)),
+        in_process: Mutex::new(None),
+        qr_code_uri,
+        ble_ident,
+    })
+}
+
+#[derive(uniffi::Object)]
+pub struct MdlPresentationSession {
+    engaged: Mutex<Option<device::SessionManagerEngaged>>,
```

Comment:
Does this need to be an Option? It seems to always be constructed with `Some`

Reply:
Yes, I just noticed the same thing. Left over from some old code, I've removed it.